services/ai_client.py

```
@@ -5,37 +5,127 @@
+        """Initialize the appropriate tokenizer for the model."""
+        try:
+            model_config = MODEL_LIMITS.get(self.model, MODEL_LIMITS["openai/llama-3.1-8b-instruct"])
+            self.tokenizer = tiktoken.get_encoding(model_config["tiktoken_model"])
+            self.model_config = model_config
+            
+            # Set max input tokens if not configured
+            if self.max_input_tokens == 0:
+                self.max_input_tokens = model_config["context_window"] - model_config["max_output_tokens"]
+                
+            logger.info(f"Initialized tokenizer for {self.model}, max input tokens: {self.max_input_tokens}")
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)